packages/apollo-federation/src/composition/utils.ts

```diff
@@ -303,3 +313,73 @@ export function selectionIncludesField({
   }
   return false;
 }
+
+export function isTypeNodeAnEntity(
+  node: TypeDefinitionNode | TypeExtensionNode,
+) {
+  let isEntity = false;
+
+  visit(node, {
+    Directive(directive) {
+      if (directive.name.value === 'key') {
+        isEntity = true;
+        return BREAK;
+      }
+    },
+  });
+
+  return isEntity;
+}
+
+export function diffTypeNodes(
```

Comment:
Can you add a comment saying exactly what this function does. I think for util files like this, it's even more important to say what a function does rathe than having to trace its usage in another file.

Reply:
Resolved in b3a34adac3cfd966e61411ca137bf00f6bb4a59f